### ISSUE TITLE
Point out which derivations are currently being built by nix daemon

### ISSUE THREAD

**Issue by nh2** [NixOS/nix#3200]
Once the nix daemon builds something, it is hard to find out what that is.

It would be great if the nix daemon could update its process/thread names to show the derivation being built.

Then one could see them in `htop`.

Thread names have a limitation to 15 chars (if I remember correctly from my discussion with @edolstra at NixCon); in case they have to be used, it would already be useful to show the first couple chars of a `.drv` which is usually enough to locate it unambiguously on the system.

**Comment by edolstra** [comment]
Note that a daemon process can be building any number of derivations. So it's not obvious which one should be displayed in the process/thread title.

**Comment by nh2** [comment]
Doesn't it fork a process, or at least a thread, per built derivation?

If not, maybe we could make it so, so that each derivation being built has a parent in the process tree between top-level nix daemon and `builder` executable that shows the `.drv`?

**Comment by edolstra** [comment]
No, the builds are handled by a giant event loop in a single thread.

Adding an extra process just for debugging sounds a bit expensive to me... Maybe if we start running builds in their own cgroup, we could name the cgroup after the derivation.

**Comment by chkno** [comment]
A crummy alternative is to [look for incomplete log files](https://github.com/chkno/nix-currently-building).

**Comment by bew** [comment]
Please no!

**Comment by jsoo1** [comment]
`nix-top` (nor any of the above methods) show remote builds.
